### DISCUSSION TITLE
Simplify usage of Success.Of

### DISCUSSION THREAD

**Discussion by tjaskula** [vlingo-net/vlingo-net-common#61] [Ideas]
The usage of `Success.Of` is cumbersome like `Success.Of<StorageException, Result>(Result.Success)`

The idea is to simplify type parameters:
- we don't need to pass in the type of the exception, by default it should have just the Exception property set if there was an issue.

It should be `Success<T>` or even better it should be `Result<T>` as "success" with a failure is more a result than a successful operation.

@VaughnVernon Any ideas?

**Comment by tjaskula** [comment]
The same can be said about `Failure<TCause, TValue>` which in that cas should be `Failure<TCause>`

**Comment by VaughnVernon** [comment]
You can put the ugly typing on the `new` inside the class method.

**Reply by tjaskula** [reply]
> You can put the ugly typing on the `new` inside the class method.

I don't understand

**Comment by VaughnVernon** [comment]
From: `Success.Of<StorageException, Result>(Result.Success) { ... }`

To: `Success.Of(Result result) { return new Success.InteralOf<StorageException, Result>(result); }`

It's only the gist, not necessarily the way.

**Reply by tjaskula** [reply]
Ok, I understand better. However in .NET implementation `StorageException` type parameter is never used in `Success` except from carrying it from one end to another. If we dropped it it could be Type of `Exception` and would be good to go.

**Comment by VaughnVernon** [comment]
Sure, that's fine. Still I think there should be an `internal` method that hides those details. Requiring someone to type an unnecessary generic declaration is a pain.
